lib/morandi/image_processor.rb

```diff
@@ -171,22 +174,20 @@ def config_for(key)
     end
 
     def apply_crop!
-      crop = options['crop']
+      return if negative_crop?
 
-      return if crop.nil? && config_for('image.auto-crop').eql?(false)
+      return if options['crop'].nil? && config_for('image.auto-crop').eql?(false)
 
-      crop = crop.split(/,/).map(&:to_i) if crop.is_a?(String) && crop =~ /^\d+,\d+,\d+,\d+/
-
-      crop = nil unless crop.is_a?(Array) && crop.size.eql?(4) && crop.all? do |i|
-        i.is_a?(Numeric)
-      end
+      crop = unpack_crop_options
 
       # can't crop, won't crop
       return if @width.nil? && @height.nil? && crop.nil?
 
-      crop = crop.map { |s| (s.to_f * @scale).floor } if crop && not_equal_to_one(@scale)
-
-      crop ||= Morandi::Utils.autocrop_coords(@pb.width, @pb.height, @width, @height)
+      if crop.nil?
+        crop ||= Morandi::Utils.autocrop_coords(@pb.width, @pb.height, @width, @height)
```

Comment:
`if crop.nil?` then it's always going to set it, on the basis it's not going to be otherwise falsey.
```suggestion
        crop = Morandi::Utils.autocrop_coords(@pb.width, @pb.height, @width, @height)
```

Reply:
Your right, I'd literally dragged the line into the if block - oop!